### PR TITLE
DateTimeInterface::ISO8601 Incompatibility with ISO8601

### PR DESCRIPTION
**Pull request by ngourle**
Related #1575

### CODE REVIEW

**Review by jderusse — COMMENTED** [review]
Tests are red because it need to be fixed in `src/CodeGenerator/src/Generator/InputGenerator.php`

File: src/CodeGenerator/src/Generator/InputGenerator.php [hunk 1]
```diff
@@ -561,7 +561,7 @@ private function stringify(string $variable, Member $member, string $part): stri
         $shape = $member->getShape();
         switch ($shape->getType()) {
             case 'timestamp':
-                $format = strtoupper($shape->get('timestampFormat') ?? ('header' === $part ? 'rfc822' : 'iso8601'));
+                $format = strtoupper($shape->get('timestampFormat') ?? ('header' === $part ? 'rfc822' : 'atom'));
```
Comment (jderusse) [inline src/CodeGenerator/src/Generator/InputGenerator.php:564]: `shape->get('timestampFormat')` might return `iso8601`.

```suggestion
                $format = strtoupper($shape->get('timestampFormat') ?? ('header' === $part ? 'rfc822' : 'atom'));
                if ($format === 'ISO8601') {
                    $format = 'ATOM'
                }
```

Comment (ngourle) [reply src/CodeGenerator/src/Generator/InputGenerator.php:564]: You are right, thank you

**Review by jderusse — COMMENTED** [review]
Last request. Could you please add an entry in the `src/Service/S3/CHANGELOG.md` file?

File: src/CodeGenerator/src/Generator/InputGenerator.php [hunk 2]
```diff
@@ -561,7 +561,11 @@ private function stringify(string $variable, Member $member, string $part): stri
...
```
Comment (stof) [inline src/CodeGenerator/src/Generator/InputGenerator.php:564]: the fallback here should still be `iso8601`, to correspond to values used in `timestampFormat` IMO. The rewriting will be done below anyway

File: src/Service/S3/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## NOT RELEASED
 
+### Fixed
+
+- According PHP documentation, format datetime with `ATOM` to provide a workaround for the incompatibility with `ISO8601` format.
```
Comment (stof) [inline src/Service/S3/CHANGELOG.md:7]: the fix should be described what is getting fixed in the library, not _how_  (as users of the library don't care about that).

So this should be ```Fix the format of the `objectLockRetainUntilDate` field in requests``` IMO

File: src/CodeGenerator/src/Generator/InputGenerator.php (hunk 2 above)
Comment (ngourle) [reply src/CodeGenerator/src/Generator/InputGenerator.php:564]: That's a good point, I'll change that quickly.

**Comment by jderusse** [comment]
thank you @ngourle

**Pull request merged**